industrial_ci/src/tests/source_tests.sh

```diff
@@ -78,6 +78,8 @@ file) # When UPSTREAM_WORKSPACE is file, the dependended packages that need to b
     elif [ -e $TARGET_REPO_PATH/$ROSINSTALL_FILENAME ]; then
         # install (maybe unreleased version) dependencies from source
         $ROSWS merge file://$TARGET_REPO_PATH/$ROSINSTALL_FILENAME
+    else
+        error "UPSTREAM_WORKSPACE file '$TARGET_REPO_PATH/$ROSINSTALL_FILENAME[.$ROS_DISTRO]' does not exist"
```

Comment:
>>    What's the purpose for enclosing the file extension as [.$ROS_DISTRO]?
>
> At this point we can't tell which file is missing, the generic or the distro-specific one.

I think I got your point; either generic or distro-specific file is missing but there's no way for the script to know which one is intended by the user.

The `[]` notation may not look the best appropriate for this case, but then I don't know what's the best. Let's just leave this as is :)